src/ipc_message.cc

```diff
@@ -29,61 +29,121 @@
 #include <memory>
 
 namespace triton { namespace backend { namespace python {
+std::unique_ptr<IPCMessage>
+IPCMessage::Create(
+    const std::unique_ptr<SharedMemoryManager>& shm_pool, bool inline_response)
+{
+  AllocatedSharedMemory<IPCMessageShm> ipc_message_shm =
+      shm_pool->Construct<IPCMessageShm>();
+
+  ipc_message_shm.data_->inline_response = inline_response;
+  AllocatedSharedMemory<bi::interprocess_mutex> response_mutex_shm;
+  AllocatedSharedMemory<bi::interprocess_condition> response_cond_shm;
+  if (inline_response) {
+    response_mutex_shm =
+        std::move(shm_pool->ConstructAligned<bi::interprocess_mutex>());
+    response_cond_shm =
+        std::move(shm_pool->ConstructAligned<bi::interprocess_condition>());
+
+    ipc_message_shm.data_->response_mutex = response_mutex_shm.handle_;
+    ipc_message_shm.data_->response_cond = response_cond_shm.handle_;
+  }
+
+  return std::unique_ptr<IPCMessage>(
+      new IPCMessage(ipc_message_shm, response_mutex_shm, response_cond_shm));
+}
 
 std::unique_ptr<IPCMessage>
 IPCMessage::LoadFromSharedMemory(
-    std::unique_ptr<SharedMemory>& shm_pool, off_t message_offset)
+    std::unique_ptr<SharedMemoryManager>& shm_pool,
+    bi::managed_external_buffer::handle_t message_offset)
```

Comment:
message_offset is no longer an offset but a handle right?

Reply:
Boost typedefs the handles to a simple number. So they are basically the same thing. But I'll rename all the offsets to handle to be consistent.